### PR TITLE
Grey and purple slime crystal fixes

### PR DESCRIPTION
**Pull request by Ravellon**
## About The Pull Request

Right now, instead of using add_nutrition() function the grey crystal directly adjusts slime's nutrition. This prevents slimes from having their life effects triggered on nutrition change, such as splitting or raising their charge, while making them unwilling to feed once nutrition rises above 1000. That makes them unable to split once they cross that threshold.

With this change, grey crystal will feed slimes to 996/1200 nutrition and 9/10 growth instead of overfeeding them.
(if the conditional checks are removed the slimes will be fed infinitely and will split at getting to 10/10 growth, which will result in infinite slime reproduction. Not a good idea, I don't think)

Also, purple crystals now will heal toxin damage for slime races instead of dealing it.

## Why It's Good For The Game

This way, leaving slimes in range of crystal for some time will put them into perfect range for either breeding or crossbreeding while not risking infinite slime spawning.
And it's just silly that xenobio healing crystals slowly kill xenobio sourced races.

## Changelog
:cl:
fix: Grey slime crystals will now slowly feed the slimes in range to 996/1200 nutrition and 9/10 growth range, instead of overfeeding them and making them unable to split.
fix: Purple slime crystals will now heal toxin damage for slime races instead of dealing it.
/:cl:

### CODE REVIEW

File: code/modules/research/xenobiology/crossbreeding/_structures.dm [hunk 1]
```diff
@@ -119,7 +119,8 @@ GLOBAL_LIST_EMPTY(bluespace_slime_crystals)
 	if(!istype(affected_mob, /mob/living/simple_animal/slime))
 		return
 	var/mob/living/simple_animal/slime/slime_mob = affected_mob
-	slime_mob.nutrition += 2
+	if((slime_mob.nutrition < 996)||(slime_mob.amount_grown < (SLIME_EVOLUTION_THRESHOLD - 1)))
```
Comment (utilbird) [inline code/modules/research/xenobiology/crossbreeding/_structures.dm]: Brackets aren't needed
```suggestion
	if(slime_mob.nutrition < 996) || slime_mob.amount_grown < SLIME_EVOLUTION_THRESHOLD - 1)
```

**Review by utilbird — CHANGES_REQUESTED** [review]

File: code/modules/research/xenobiology/crossbreeding/_structures.dm [hunk 2]
```diff
@@ -119,7 +119,8 @@ GLOBAL_LIST_EMPTY(bluespace_slime_crystals)
 	if(!istype(affected_mob, /mob/living/simple_animal/slime))
 		return
 	var/mob/living/simple_animal/slime/slime_mob = affected_mob
-	slime_mob.nutrition += 2
+	if(slime_mob.nutrition < 996) || slime_mob.amount_grown < SLIME_EVOLUTION_THRESHOLD - 1)
```
Comment (utilbird) [inline code/modules/research/xenobiology/crossbreeding/_structures.dm]: Ah crap, messed up the initial code suggestion lmao, my bad

```suggestion
	if(slime_mob.nutrition < 996 || slime_mob.amount_grown < SLIME_EVOLUTION_THRESHOLD - 1)
```

**Comment by Ravellon** [comment]
I missed it too, so it's my fault for not being attentive enough.

Having no experience with BYOND I just do what has to work for sure no matter the language.

**Comment by Ravellon** [comment]
On a related note. The purple crystal is very very weak. It has 1/10 chance to heal 2 damage of any basic type or 1/10 chance to heal 2 damage to heart, lungs, liver or brain.
And it does not matter if there is no damage of the chosen type, it tries to heal it anyway and wastes the healing tick.

This is just garbage, tbh.

**Comment by Rukofamicom** [comment]
> This prevents slimes from having their life effects triggered on nutrition change, such as splitting

This is not true at all. Slimes fed by the crystal grow and reproduce just the same as normal. I tested this crystal by putting it in an enclosure with one slime while doing all the rest of the bugtesting. By the time I was done with the rest of the bugtesting, the pen had 8-12 slimes per tile.

No electric charge is probably correct though. Not being able to feed is negligible since they divide anyway. Are you certain you hadn't reached the entity limit when you had your experience? This will prevent slimes from feeding and reproducing

**Comment by Ravellon** [comment]
This is very weird. I managed to reproduce my issue once. The other three times it was as you described.
As in, one time they stuck at under 10 growth and got overfed, making them unable to split, even after I culled their numbers.
The other three times they continued to split after I culled the numbers.

Closing for now.

**Pull request closed without merging**